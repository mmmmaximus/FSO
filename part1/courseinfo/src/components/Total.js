import React from 'react'

const Total = (props) => {
  let total = 0
  for (let i=0; i < props.parts.length; i++) {
    total += props.parts[i].exercises
  }

  return (
    <p>
      <strong>
        total of {total} exercises
      </strong>
    </p>
  )
}

export default Total
