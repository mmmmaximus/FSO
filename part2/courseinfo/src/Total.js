import React from 'react'

const Total = (props) => {
  let total = 0
  for (let i=0; i < props.parts.length; i++) {
    total += props.parts[i].exercises
  }

  return (
    <p>
      Number of exercises {total}
    </p>
  )
}

export default Total
