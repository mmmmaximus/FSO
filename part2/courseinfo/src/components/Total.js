import React from 'react'

const Total = ({parts}) => {
  const total = parts.map(part => part.exercises).reduce((a,b) => a+b)

  return (
    <p>
      <strong>
        total of {total} exercises
      </strong>
    </p>
  )
}

export default Total
