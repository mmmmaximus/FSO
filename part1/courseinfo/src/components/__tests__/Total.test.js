import { render, screen } from '@testing-library/react'
import Total from '../Total'
import renderer from 'react-test-renderer'

it('should render Total component', () => {
  const course = {
    name: 'Half Stack application development',
    parts: [
      {
        name: 'Fundamentals of React',
        exercises: 10
      },
      {
        name: 'Using props to pass data',
        exercises: 7
      },
      {
        name: 'State of a component',
        exercises: 14
      }
    ]
  }
  // check render properly
  render(<Total parts={course.parts}/>)
  const TotalElement = screen.getByTestId('Total')
  expect(TotalElement).toBeInTheDocument

  // check Total content
  let total = 0
  for (let i=0; i < course.parts.length; i++) {
    total += course.parts[i].exercises
  }
  expect(TotalElement.textContent).toEqual(`Number of exercises ${total}`)
})

it('matches snapshot', () => {
  const course = {
    name: 'Half Stack application development',
    parts: [
      {
        name: 'Fundamentals of React',
        exercises: 10
      },
      {
        name: 'Using props to pass data',
        exercises: 7
      },
      {
        name: 'State of a component',
        exercises: 14
      }
    ]
  }
  const tree = renderer.create(<Total parts={course.parts}/>).toJSON()
  expect(tree).toMatchSnapshot()
})
