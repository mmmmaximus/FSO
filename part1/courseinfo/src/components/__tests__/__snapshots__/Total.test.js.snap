// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<p
  data-testid="Total"
>
  Number of exercises 
  31
</p>
`;
